examples/aks/stack.tf

```
@@ -0,0 +1,192 @@
+    storage_account                  = azurerm_storage_account.this.name
+    managed_identity_node_rg_name    = module.cluster.node_resource_group
+    managed_identity_oidc_issuer_url = module.cluster.oidc_issuer_url
+  }
+
+  dependency_ids = {
+    azure-workload-identity = module.azure-workload-identity.id
+  }
+}
+
+module "thanos" {
+  source = "git::https://github.com/camptocamp/devops-stack-module-thanos.git//aks?ref=v1.0.0"
+
+  cluster_name     = local.cluster_name
+  base_domain      = azurerm_dns_zone.this.name
+  argocd_namespace = module.argocd.argocd_namespace
+  cluster_issuer   = "letsencrypt-staging"
+
+  metrics_storage = {
+    container           = azurerm_storage_container.metrics.name
+    storage_account     = azurerm_storage_account.this.name
+    storage_account_key = azurerm_storage_account.this.primary_access_key
+  }
+
+  thanos = {
+    oidc = local.oidc
+  }
+
+  dependency_ids = {
+    cert-manager = module.cert-manager.id
+  }
+}
+
+module "monitoring" {
+  source = "git::https://github.com/camptocamp/devops-stack-module-kube-prometheus-stack.git//aks?ref=v2.3.0"
+
+  cluster_name     = local.cluster_name
+  base_domain      = azurerm_dns_zone.this.name
+  argocd_namespace = module.argocd.argocd_namespace
+  cluster_issuer   = "letsencrypt-staging"
+
+  alertmanager = {
+    oidc = local.oidc
+  }
+  prometheus = {
+    oidc = local.oidc
+  }
+  grafana = {
+    oidc                    = local.oidc
+    additional_data_sources = true
+  }
+
+  metrics_storage = {
+    container           = azurerm_storage_container.metrics.name
+    storage_account     = azurerm_storage_account.this.name
+    storage_account_key = azurerm_storage_account.this.primary_access_key
+  }
+
+  dependency_ids = {
+    cert-manager = module.cert-manager.id
+  }
+}
+
+module "argocd_final" {
+  source = "git::https://github.com/camptocamp/devops-stack-module-argocd.git?ref=v1.1.0"
+
+  cluster_name   = local.cluster_name
+  base_domain    = azurerm_dns_zone.this.name
+  cluster_issuer = "letsencrypt-staging"
+
+  admin_enabled            = "true"
+  namespace                = module.argocd.argocd_namespace
+  accounts_pipeline_tokens = module.argocd.argocd_accounts_pipeline_tokens
+  server_secretkey         = module.argocd.argocd_server_secretkey
+
+  oidc = {
+    name            = "OIDC"
+    issuer          = local.oidc.issuer_url
+    clientID        = local.oidc.client_id
+    clientSecret    = local.oidc.client_secret
+    requestedScopes = ["openid", "profile", "email"]
+    requestedIDTokenClaims = {
+      groups = {
+        essential = true
+      }
+    }
+  }
+
+  helm_values = [{
+    argo-cd = {
+      configs = {
+        rbac = {
+          "policy.csv" = <<-EOT
+          g, pipeline, role:admin
+          g, argocd-admin, role:admin
+          EOT
+        }
+      }
+    }
+  }]
+
+  dependency_ids = {
+    kube-prometheus-stack = module.monitoring.id
+    cert-manager          = module.cert-manager.id
```

```suggestion
    traefik               = module.traefik.id
    cert-manager          = module.cert-manager.id
    kube-prometheus-stack = module.kube-prometheus-stack.id
```

I think we should explicitly declare all modules that this module depends on an not depend on the implicit dependencies throughout the deployment.